smartsim/_core/_cli/scripts/dragon_install.py

```diff
@@ -152,40 +267,56 @@ def retrieve_asset_info() -> GitReleaseAsset:
     return asset
 
 
-def retrieve_asset(working_dir: pathlib.Path, asset: GitReleaseAsset) -> pathlib.Path:
+def retrieve_asset(
+    request: DragonInstallRequest, asset: GitReleaseAsset
+) -> pathlib.Path:
```

Comment:
Oh and can you also add a `:raises:` docstring?

Reply:
I can!